### DISCUSSION TITLE
New save file format

### DISCUSSION THREAD

**Discussion by Try** [Try/OpenGothic#167] [General]
I have an idea to completely rework save-game files, because current is good, but not great. Here is some thoughts:

Current file format is tightly packed binary file. Structure:
```
{
  header, version,
  current world data,
  other_worlds:
    [
      {
       version,
       world data,
      },
     other visited worlds
    ]
}
```
Pros: simple, compact
Cons: already 30 different versions; tracking issues pretty much impossible.

#### New format
Idea is to build format on top of .zip, so it can be viewed in regular app, like 7z and be compact in the same time 

Pros: compact, somewhat-human readable, save game can be fixed, if needed.
Cons: no compatibility with previous *.sav

#### When
If accepted, change will be implemented **after** next release

#### Question
Want to know, what do you think about this change and will compatibility drop be devastating or acceptable?

**Comment by ernierasta** [comment]
I do not know, how many people has some significant progress under OpenGothic. From my point of view, it is reasonable to change it now, because later it could be serious problem and would require to write some kind of internal or external save converter. Personally I think this change would be good, especially for development/debugging purpose. While plain text saves makes it easier to cheat, I am not sure if in context of Gothic this is problem, probably not.

**Comment by BlackMagicCoding** [comment]
In a single player game it is always nice to be able to edit a save game if problems arise, or just to have some fun with it.
I would change it with the same arguments as ernierasta already said :)

**Comment by Abendlied** [comment]
To me it appears advantageous to be able to manipulate the savegame by hand, also for testing purposes. In my view this advantage would even justify slight performance drops while saving.
What is the possible impact on mod compatibility, will this be an additional burden? If not I can only echo ernierasta, rather to go through with the transition now than later.

**Comment by Try** [comment]
This is how new save-game file look a like from the inside:
![изображение](https://user-images.githubusercontent.com/122894/137797621-2f251aac-1603-4c9a-ad98-4e515078d77c.png)

* Can be open/changed by 7zip or WinRar or other similar app
* Information about npc-state and npc-animation is split, to make animation work easier in the future
* In-game world are represented as separated sub-folders, to make it easier to debug invalid save-games

**Comment by Try** [comment]
Game build(1311) with new save-format was released
